File: app/src/main/java/cz/krystofcejchan/upocasi/objects/Preferences.java
package cz.krystofcejchan.upocasi.objects;

import android.content.Context;
import android.content.SharedPreferences;

import androidx.preference.PreferenceManager;

public class Preferences {
    private final SharedPreferences sharedPreferences;

    public Preferences(Context context) {
        sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
    }

    public boolean isCelsiusDefault() {
        return sharedPreferences.getString("temp_unit", "celsius").equals("celsius");
    }

    public String getDefaultLocation() {
        return sharedPreferences.getString("location", "Olomouc");
    }

    public SharedPreferences getSharedPreferences() {
        return sharedPreferences;
    }
}

File: app/src/main/java/cz/krystofcejchan/upocasi/objects/Preferences.kt
package cz.krystofcejchan.upocasi.objects

import android.content.Context
import android.content.SharedPreferences
import androidx.preference.PreferenceManager

class Preferences(context: Context?) {
    private val sharedPreferences: SharedPreferences

    init {
        sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context!!)
    }

    val isCelsiusDefault: Boolean
        get() = sharedPreferences.getString("temp_unit", "celsius") == "celsius"
    val defaultLocation: String?
        get() = sharedPreferences.getString("location", "Olomouc")
}